src/internal/CallServiceHandler.ts

```diff
@@ -82,6 +85,20 @@ export interface CallServiceResult {
  */
 export type Middleware = (req: CallServiceData, resp: CallServiceResult, next: Function) => void;
 
+export class CallServiceArg<T> {
+    val: T;
+    tetraplet: SecurityTetraplet[];
+
+    constructor(val: T, tetraplet: SecurityTetraplet[]) {
+        this.val = val;
+        this.tetraplet = tetraplet;
+    }
+}
+
+type CallParams = ParticleContext & {
+    wrappedArgs: CallServiceArg<any>[];
```

Comment:
Why `<any>`? Could it be a concrete type?

If not, why have a generic `<T>` if it's used as `any`?

Reply:
In theory it is possible to replace `any` with the unknown. But it will might break the compatibility with little bits where the old API is still used (e.g fldist). 

This will be removed after the adoption of the new SDK API